### PR TITLE
Triggeralgs v4 to v5 port

### PR DESCRIPTION
**Pull request by ArturSztuc**
Porting trigger algorithms from v4 to v5.
Full description in  https://github.com/DUNE-DAQ/triggeralgs/pull/78.

Has to be merged together with https://github.com/DUNE-DAQ/triggeralgs/pull/78, https://github.com/DUNE-DAQ/trigger/pull/339, https://github.com/DUNE-DAQ/trgdataformats/pull/41 and https://github.com/DUNE-DAQ/fdreadoutlibs/pull/220.

### CODE REVIEW

File: schema/appmodel/trigger.schema.xml [hunk 1]
```diff
@@ -247,11 +247,12 @@
  <class name="TAMakerADCSimpleWindowAlgorithm">
   <superclass name="TAAlgorithm"/>
   <attribute name="adc_threshold" description="Threshold for the ADC integral algorithm" type="u32" init-value="10000" is-not-null="yes"/>
-  <attribute name="window_length" description="Window length (in ticks) for the ADC integral algorithm" type="u32" init-value="10000" is-not-null="yes"/>
+  <attribute name="window_length" description="Window length (in ticks) for the algorithm" type="u32" init-value="10000" is-not-null="yes"/>
```
Comment (eflumerf) [inline schema/appmodel/trigger.schema.xml:250]: Is window_length a property of all TAAlgorithms? Should this attribute be moved to the base class?

Comment (ArturSztuc) [reply schema/appmodel/trigger.schema.xml:250]: Hi @eflumerf, it is not. Most of our algorithms do use a window_length, but not all of them (e.g. bundle algorithm, prescale algorithm, DBSCAN too I think), which is why I didn't add it to the base class.

Comment (aeoranday) [reply schema/appmodel/trigger.schema.xml:250]: @ArturSztuc Could there be another class level, e.g., `TAWindowAlgorithm` that is derived from `TAAlgorithm` and has `window_length`?

Comment (ArturSztuc) [reply schema/appmodel/trigger.schema.xml:250]: @aeoranday I actually don't know, but I imagine this should be possible

**Review by aeoranday — COMMENTED** [review]
I notice many of the TCMakers are missing. Was this an oversight?

**Comment by ArturSztuc** [comment]
@aeoranday That was definitely an oversight, I fixed it now in the last 2 commits.

**Review by aeoranday — APPROVED** [review]
LGTM.

Tested all related PRs by going through a test-session for a few different TAMaker and TCMaker configurations. All produced meaningful TAs/TCs, and the debug BundleN makers produced consistent counts for their configuration (reminds me to port `trgtools`).

**Pull request merged**